### DISCUSSION TITLE
Version 2.1 crashes the BE process directly when using count(distinct column_name) in multi-table joins.

### DISCUSSION THREAD

**Discussion by joeykoer** [apache/doris#32389] [A - General / Q&A]
query:
```
select a.c_store_id,c.cate_id,count(distinct a.c_id)
from ODS.store_sale a left join ODS.os_dim_goods_v1_001 c on a.c_gcode=c.goods_code
where a.c_datetime>='2024-02-27'
and a.c_datetime<'2024-02-28'
and a.c_id not like '-%'
group by a.c_store_id,c.cate_id
```

After upgrading from 2.0.5 to 2.1.0, such a query causes all associated BEs to crash. There are no obvious errors in the BE logs, only the FE reports that the query has been cancelled. The same statement was functioning normally in previous versions. 

Adding count(1) allows the query to run normally.
```
-- add "count(1)" 
select a.c_store_id,c.cate_id,count(1),count(distinct a.c_id) sheet_num
from ODS.store_sale a left join ODS.os_dim_goods_v1_001 c on a.c_gcode=c.goods_code
where a.c_datetime>='2024-02-27'
and a.c_datetime<'2024-02-28'
and a.c_id not like '-%'
group by a.c_store_id,c.cate_id
```

**Comment by yagagagaga** [comment]
This looks like an OOM occurred with be. You can execute `dmesg -T` in Linux that BE has deployed for more message.

**Reply by jiamxie11** [reply]
the same problem, BE process crashes when using count(DISTINCT dt) at both version 2.1 and 2.0.5

SQL like:
            SELECT
                coalesce(dim_businessline, '') AS dim_businessline,
                sum(order_num) / count(DISTINCT dt) AS order_num
                 ...
            FROM
                xxxyyy
            WHERE
                (
                    dt BETWEEN '2024-03-12' AND '2024-03-18'
                )
            GROUP BY
                GROUPING SETS((coalesce(dim_businessline, '')), ());

be.out

start time: Wed Mar 20 11:22:06 CST 2024
INFO: java_cmd /opt/programs/java/bin/java
INFO: jdk_version 8
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/opt/programs/doris/apache-doris-2.1.0-bin-x64/be/lib/java_extensions/preload-extensions/preload-extensions-jar-with-dependencies.jar!/org/slf4j/impl/St
aticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/opt/programs/doris/apache-doris-2.1.0-bin-x64/be/lib/java_extensions/java-udf/java-udf-jar-with-dependencies.jar!/org/slf4j/impl/StaticLoggerBinder.cla
ss]
SLF4J: Found binding in [jar:file:/opt/programs/doris/apache-doris-2.1.0-bin-x64/be/lib/hadoop_hdfs/common/lib/slf4j-reload4j-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Reload4jLoggerFactory]
[WARNING!] /sys/kernel/mm/transparent_hugepage/enabled: [always] madvise never, Doris not recommend turning on THP, which may cause the BE process to use more memory and cannot be freed 
in time. Turn off THP: `echo madvise | sudo tee /sys/kernel/mm/transparent_hugepage/enabled`
*** Query id: b5665b7f1d794d8d-8cdf003416e6f953 ***
*** tablet id: 0 ***
*** Aborted at 1710904949 (unix time) try "date -d @1710904949" if you are using GNU date ***
*** Current BE git commitID: 91efb6a43d ***
*** SIGSEGV address not mapped to object (@0x0) received by PID 43702 (TID 46416 OR 0x7f0bb7765700) from PID 0; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /home/zcp/repo_center/doris_release/doris/be/src/common/signal_handler.h:417
 1# os::Linux::chained_handler(int, siginfo*, void*) in /opt/programs/java/jre/lib/amd64/server/libjvm.so
 2# JVM_handle_linux_signal in /opt/programs/java/jre/lib/amd64/server/libjvm.so
 3# signalHandler(int, siginfo*, void*) in /opt/programs/java/jre/lib/amd64/server/libjvm.so
 4# 0x00007F10D5B9E400 in /lib64/libc.so.6
 5# memcpy at /home/zcp/repo_center/doris_release/doris/be/src/glibc-compatibility/memcpy/memcpy_x86_64.cpp:219
 6# doris::vectorized::ColumnString::insert_range_from(doris::vectorized::IColumn const&, unsigned long, unsigned long) in /opt/programs/doris/be/lib/doris_be
 7# doris::vectorized::ColumnNullable::insert_range_from(doris::vectorized::IColumn const&, unsigned long, unsigned long) at /home/zcp/repo_center/doris_release/doris/be/src/vec/columns/
column_nullable.cpp:282
 8# doris::pipeline::RepeatLocalState::get_repeated_block(doris::vectorized::Block*, int, doris::vectorized::Block*) in /opt/programs/doris/be/lib/doris_be
 9# doris::pipeline::RepeatOperatorX::pull(doris::RuntimeState*, doris::vectorized::Block*, bool*) const at /home/zcp/repo_center/doris_release/doris/be/src/pipeline/exec/repeat_operator
.cpp:221
10# doris::pipeline::StatefulOperatorX<doris::pipeline::RepeatLocalState>::get_block(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /home/zcp/repo_center/doris_release/doris/
be/src/pipeline/pipeline_x/operator.cpp:459
11# doris::pipeline::OperatorXBase::get_block_after_projects(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /home/zcp/repo_center/doris_release/doris/be/src/pipeline/pipeline
_x/operator.cpp:210
12# doris::pipeline::StatefulOperatorX<doris::pipeline::StreamingAggLocalState>::get_block(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /home/zcp/repo_center/doris_release/
doris/be/src/pipeline/pipeline_x/operator.cpp:444
13# doris::pipeline::OperatorXBase::get_block_after_projects(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /home/zcp/repo_center/doris_release/doris/be/src/pipeline/pipeline
_x/operator.cpp:210
14# doris::pipeline::PipelineXTask::execute(bool*) at /home/zcp/repo_center/doris_release/doris/be/src/pipeline/pipeline_x/pipeline_x_task.cpp:274
15# doris::pipeline::TaskScheduler::_do_work(unsigned long) at /home/zcp/repo_center/doris_release/doris/be/src/pipeline/task_scheduler.cpp:334
16# doris::ThreadPool::dispatch_thread() in /opt/programs/doris/be/lib/doris_be
17# doris::Thread::supervise_thread(void*) at /home/zcp/repo_center/doris_release/doris/be/src/util/thread.cpp:499
18# start_thread in /lib64/libpthread.so.0
19# __clone in /lib64/libc.so.6

Is there any workaroud? 
thanks!
